contrib/export/v036/gov/types.go

```diff
@@ -0,0 +1,43 @@
+package gov
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/gov"
+	"time"
+)
+
+// TODO: import from x/gov all the types that have not changed
+
+// GenesisState - all staking state that must be provided at genesis
+type GenesisState struct {
+	StartingProposalID uint64            `json:"starting_proposal_id"`
+	Deposits           gov.Deposits      `json:"deposits"`
+	Votes              gov.Votes         `json:"votes"`
+	Proposals          []Proposal        `json:"proposals"`
+	DepositParams      gov.DepositParams `json:"deposit_params"`
+	VotingParams       gov.VotingParams  `json:"voting_params"`
+	TallyParams        gov.TallyParams   `json:"tally_params"`
+}
+
+// Param around deposits for governance
+type DepositParams struct {
+	MinDeposit       sdk.Coins     `json:"min_deposit,omitempty"`        //  Minimum deposit for a proposal to enter voting period.
+	MaxDepositPeriod time.Duration `json:"max_deposit_period,omitempty"` //  Maximum period for Atom holders to deposit on a proposal. Initial value: 2 months
+}
+
+type Proposal struct {
+	gov.Content `json:"content"` // Proposal content interface
+
+	ProposalID       uint64             `json:"id"`                 //  ID of the proposal
+	Status           gov.ProposalStatus `json:"proposal_status"`    // Status of the Proposal {Pending, Active, Passed, Rejected}
```

Comment:
Dangerous to use any types directly from the module as they may change. May we move these in here?

Reply:
More or less, if I do so the migration function will become much more verbose. My first version was a dump of everything related to gov, but when I started writing the Migrate func it became a continuous type clash.

I agree! indeed it's safer, but I felt like using the native module it's helping in understanding exaclty what's changing